src/time_bucket_ng.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#include <postgres.h>
+#include <utils/date.h>
+#include <utils/datetime.h>
+#include <utils/fmgrprotos.h>
+
+#include "time_bucket_ng.h"
+
+TS_FUNCTION_INFO_V1(ts_time_bucket_ng);
+TS_FUNCTION_INFO_V1(ts_time_bucket_ng_timestamp);
+TS_FUNCTION_INFO_V1(ts_time_bucket_ng_timestamptz);
+
+TSDLLEXPORT Datum
+ts_time_bucket_ng_timestamp(PG_FUNCTION_ARGS)
+{
+	DateADT result;
+	Datum interval = PG_GETARG_DATUM(0);
+	DateADT ts_date = DatumGetDateADT(DirectFunctionCall1(timestamp_date, PG_GETARG_DATUM(1)));
```

Comment:
Is this really the right thing to do given the loss of information? A Date type is only day, so losing hours, minutes, etc.

Reply:
This is a temporary implementation, the code will be rewritten in the future. Please see my more full answer in the thread https://github.com/timescale/timescaledb/pull/3211#issuecomment-864057030